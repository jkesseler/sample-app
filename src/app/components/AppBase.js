import React from 'react';
import PropTypes from 'prop-types';
import CssBaseline from '@material-ui/core/CssBaseline';
import Layout from './Layout';


// TODO: Add Header, Footer and navigation
const AppBase = ({ children }) => (
  <CssBaseline>
    <Layout>
      { children }
    </Layout>
  </CssBaseline>
);

AppBase.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default AppBase;
