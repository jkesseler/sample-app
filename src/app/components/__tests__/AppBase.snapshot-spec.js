import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import AppBase from '../AppBase';

const Child = () => (<span>fake child</span>);

describe('AppBase', () => {
  it('renders without crashing', () => {
    const component = shallow(<AppBase><Child /></AppBase>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
