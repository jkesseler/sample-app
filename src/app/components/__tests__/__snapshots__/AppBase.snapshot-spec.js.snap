// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBase renders without crashing 1`] = `
<WithStyles(CssBaseline)>
  <React.Memo>
    <Child />
  </React.Memo>
</WithStyles(CssBaseline)>
`;
